.github/workflows/check-for-updates.yml

```diff
@@ -32,3 +34,30 @@ jobs:
             -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
             https://api.github.com/repos/${{ github.repository }}/actions/workflows/nightly-build.yml/dispatches \
             -d '{"ref":"main"}'
+
+  # Checks the status of the last workflow (nightly_build) using the GitHub actions REST API and checks if the 'conclusion' state was 'success' or not (used to fail this workflow)
+  check_status:
+    runs-on: ubuntu-latest
+    needs: [ verify_commit, trigger_build ]
+    if: ${{ needs.verify_commit.outputs.RUN_BUILD == 'false' }}
+    steps:
+      - name: Check last workflow status
+        id: check_status
+        run: |
+          workflow_filename="nightly-build.yml"
+          last_workflow=$(curl -s -L -H "Accept: application/vnd.github+json" \    
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            "https://api.github.com/repos/${{ github.repository }}/actions/workflows/$workflow_filename/runs?per_page=1" | jq -r '.workflow_runs[0]')
+          conclusion=$(echo $last_workflow | jq -r '.conclusion')
+          echo "conclusion=$conclusion" >> $GITHUB_ENV
+
+      - name: Print and handle the status
+        run: |
+          echo "The status of the last workflow run is: ${{ env.conclusion }}"
+          if [ "${{ env.conclusion }}" != "success" ]; then
+            echo "The last workflow did not succeed. Failing this workflow."
+            exit 1
+          else
+            echo "The last workflow succeeded. This workflow will succeed."
+          fi
```

Comment:
We do not need this complex stuff anymore:
The check-for-updates workflow will run every night and will always be green.
The nightly-build will run whenever there had been changes and show the proper build status.
The readme badge will show the status of the last nightly build what is exactly what we want to have.

```suggestion
```